shared/actions/chat/index.js

```diff
@@ -1321,39 +1273,90 @@ function _exitSearch(
   ])
 }
 
-const _setNotifications = function*(action: Constants.SetNotifications) {
+const _setNotifications = function*(
+  action:
+    | Constants.SetNotifications
+    | Constants.ToggleChannelWideNotifications
+    | Constants.UpdatedNotifications
+) {
   const {payload: {conversationIDKey}} = action
-  // Send the new post-reducer setNotifications structure to the service.
-  const inbox = yield select(Shared.selectedInboxSelector, conversationIDKey)
-  if (inbox && inbox.notifications) {
-    const {notifications} = inbox
-    const param = {
-      convID: Constants.keyToConversationID(conversationIDKey),
-      channelWide: notifications.channelWide,
-      settings: [
-        {
-          deviceType: CommonDeviceType.desktop,
-          kind: ChatTypes.CommonNotificationKind.atmention,
-          enabled: notifications.desktop.atmention,
-        },
-        {
-          deviceType: CommonDeviceType.desktop,
-          kind: ChatTypes.CommonNotificationKind.generic,
-          enabled: notifications.desktop.generic,
-        },
-        {
-          deviceType: CommonDeviceType.mobile,
-          kind: ChatTypes.CommonNotificationKind.atmention,
-          enabled: notifications.mobile.atmention,
-        },
-        {
-          deviceType: CommonDeviceType.mobile,
-          kind: ChatTypes.CommonNotificationKind.generic,
-          enabled: notifications.mobile.generic,
-        },
-      ],
+
+  // update the one in the store
+  const old = yield select(s => s.entities.inbox.get(conversationIDKey))
+  if (old) {
+    let nextNotifications = {}
+
+    if (action.type === 'chat:setNotifications') {
+      const {payload: {deviceType, notifyType}} = action
+
+      // This is the flip-side of the logic in the notifications container.
+      if (old.notifications && old.notifications[deviceType]) {
```

Comment:
Why do you have to check for deviceType here?

Reply:
this is just pulled from the reducer where it existed before